src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalyzer.cs

```diff
@@ -23,12 +23,12 @@ public abstract class DiagnosticAnalyzer
 
         public sealed override bool Equals(object obj)
         {
-            return ReferenceEqualityComparer.Instance.Equals(this, obj);
+            return ReferenceEquals(this, obj);
         }
 
         public sealed override int GetHashCode()
         {
-            return ReferenceEqualityComparer.Instance.GetHashCode(this);
+            return ReferenceEqualityComparer.GetHashCode(this);
```

Comment:
I think this way it is more apparent we do GetHashCode as for a reference. 

I do not feel very strongly about this, but base.GetHashCode and RuntimeHelpers would seem to feel a bit magical for less familiar reader.